tests/test_thermostat.py

```diff
@@ -137,7 +137,10 @@ def test_get_data(self):
             "sessionid": api._session_id,
             "serialnumber": serial_number
         }
-        request_url = "{}?{}".format(config.THERMOSTAT_URL, urlencode(params))
+        request_url = "{}?{}".format(
+            util.get_thermostat_url(),
```

Comment:
and these would obviously all be...

```suggestion
            thermostat._url,
```

Reply:
I like it. I've got the refactor going, and I agree that it's cleaner, but unfortunately it causes a circular dependency in testing because of how `NuHeatThermostat.__init__()` calls `self.get_data()`. This means that a test response must be created before the thermostat instance is initialized, but the URL isn't available for that until it's initialized. 

What I've done in my next commit is add a helper class method called `NuHeatThermostat.get_url()` (name suggestions welcome, I guess I'm not feeling creative at the moment) to use in tests so that we can get the thermostat URL after initializing the NuHeat API object with its URL but before initializing the NuHeatThermostat one. 

Let me know what you think.

Reply:
ahhh, interesting. good catch. total oversight of mine.

since it only affects the testing mocks, we can just update the tests to build the string rather than add a helper method to the class simply to serve the tests.

e.g.
```
        responses.add(
            responses.GET,
            config.THERMOSTAT_URL,
            f"{api._api_url}/thermostat",
            status=200,
            body=json.dumps(response_data),
            content_type="application/json"
        )
```

that or make a helper in the testing library...

sorry for the super asynchronous back and forth on this. i may pull your branch and play around a bit.

Reply:
i pulled your branch, removed that test helper, and added a couple `parameterized` brand tests https://github.com/broox/python-nuheat/pull/14

really appreciate your help on this. let me know what you think!